gdk4-win32/src/win32_display.rs

```diff
@@ -24,4 +25,74 @@ impl Win32Display {
             }
         }
     }
+
+    #[doc(alias = "gdk_win32_display_add_filter")]
+    pub fn add_filter<F>(&self, filter_func: F) -> Win32DisplayFilterHandle
+    where
+        F: Fn(&Self, &mut MSG, &mut i32) -> Win32MessageFilterReturn + 'static,
+    {
+        let box_ = Box::into_raw(Box::new(filter_func)) as *mut _;
+        let func = unsafe {
+            let func: ffi::GdkWin32MessageFilterFunc = Some(trampoline::<F>);
+            ffi::gdk_win32_display_add_filter(self.to_glib_none().0, func, box_);
+            func
+        };
+
+        let display = glib::WeakRef::new();
+        display.set(Some(self));
+
+        let drop_ = |b| unsafe {
+            let _ = Box::<F>::from_raw(b as *mut _);
+        };
+        Win32DisplayFilterHandle {
+            display,
+            func,
+            box_,
+            drop_,
+        }
+    }
+}
+
+unsafe extern "C" fn trampoline<
+    F: Fn(&Win32Display, &mut MSG, &mut i32) -> Win32MessageFilterReturn + 'static,
+>(
+    display: *mut ffi::GdkWin32Display,
+    msg: glib::ffi::gpointer,
+    return_value: *mut libc::c_int,
+    box_: glib::ffi::gpointer,
+) -> i32 {
+    let f: &F = &*(box_ as *const F);
+    f(
+        &from_glib_borrow(display as *mut ffi::GdkWin32Display),
+        &mut *(msg as *mut MSG),
+        &mut *return_value,
+    )
+    .into_glib()
+}
+
+/// An owned `Win32Display` filter.
```

Comment:
```suggestion
// rustdoc-stripper-ignore-next
/// An owned `Win32Display` filter.
```
otherwise the CI will fail